bug: provider crash when merging body and sensitive body

Hi @ms-henglu 

I am experiencing a provider crash when trying to create an AKS resource. The configuration uses `sensitive_body` to conditionally add a Windows admin password when the AKS cluster is deployed in Windows mode. The crash happens when the cluster is NOT being deployed in Windows mode:

### Stack trace

```text
module.automatic.azapi_resource.this: Creating...
╷
│ Error: Request cancelled
│
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azapi_v2.7.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 66 [running]:
github.com/Azure/terraform-provider-azapi/internal/services.(*AzapiResource).CreateUpdate(0x14001d52038, {0x1157d05b8, 0x140022e2d80}, {{{0x1157d6378, 0x1400086b200}, {0x115693c60, 0x14000842ff0}}, {0x1157d9bf0, 0x14000050af0}}, {{{0x1157d6378, ...}, ...}, ...}, ...)
        github.com/Azure/terraform-provider-azapi/internal/services/azapi_resource.go:742 +0x21ac
github.com/Azure/terraform-provider-azapi/internal/services.(*AzapiResource).Create(0x100000000?, {0x1157d05b8?, 0x140022e2d80?}, {{{{0x1157d6378, 0x1400086b200}, {0x115693c60, 0x14000842ff0}}, {0x1157d9bf0, 0x14000050af0}}, {{{0x1157d6378, ...}, ...}, ...}, ...}, ...)
        github.com/Azure/terraform-provider-azapi/internal/services/azapi_resource.go:645 +0x7c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x140001f6b48, {0x1157d05b8, 0x140022e2d80}, 0x140008294f8, 0x140008294d0)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_createresource.go:101 +0x3fc
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140001f6b48, {0x1157d05b8, 0x140022e2d80}, 0x140022e8460, 0x140008295f0)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_applyresourcechange.go:57 +0x380
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140001f6b48, {0x1157d05b8?, 0x140022e2c90?}, 0x140022e83c0)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_applyresourcechange.go:55 +0x2dc
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000232500, {0x1157d05b8?, 0x140022e21b0?}, 0x140022ee070)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:866 +0x294
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x11579ba20, 0x14000232500}, {0x1157d05b8, 0x140022e21b0}, 0x14000330580, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000210400, {0x1157d05b8, 0x140022e2150}, 0x14001b9e060, 0x140002ef4d0, 0x115e807d8, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xc38
google.golang.org/grpc.(*Server).handleStream(0x14000210400, {0x1157d14c8, 0x1400022c340}, 0x14001b9e060)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 14
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x13
```

### Plan output

You will see here that I am supplying a `sensitive_body_version` for the windows admin password, but since we are not using it the value is null.

```text
 # module.automatic.azapi_resource.this will be created
  + resource "azapi_resource" "this" {
      + body                      = {
          + properties = (known after apply)
          + sku        = {
              + name = "Automatic"
              + tier = "Standard"
            }
        }
      + create_headers            = {
          + "User-Agent" = "fork_avm=true random_id=3cb445cd-bf8b-f285-0a62-0a55a6cac7a7"
        }
      + delete_headers            = {
          + "User-Agent" = "fork_avm=true random_id=3cb445cd-bf8b-f285-0a62-0a55a6cac7a7"
        }
      + id                        = (known after apply)
      + ignore_casing             = false
      + ignore_missing_property   = true
      + ignore_null_property      = false
      + location                  = "eastasia"
      + name                      = "aks-k4hb"
      + output                    = (known after apply)
      + parent_id                 = "/subscriptions/57934baa-70ec-4d1a-95e0-3799a838811f/resourceGroups/rg-k4hb"
      + read_headers              = {
          + "User-Agent" = "fork_avm=true random_id=3cb445cd-bf8b-f285-0a62-0a55a6cac7a7"
        }
      + response_export_values    = [
          + "properties.oidcIssuerProfile.issuerURL",
          + "properties.identityProfile",
          + "properties.nodeResourceGroup",
          + "properties.fqdn",
        ]
      + schema_validation_enabled = false
      + sensitive_body            = (write-only attribute)
      + sensitive_body_version    = {
          + "properties.windowsProfile.adminPassword" = null
        }
      + tags                      = (known after apply)
      + type                      = "Microsoft.ContainerService/managedClusters@2025-07-01"
      + update_headers            = {
          + "User-Agent" = "fork_avm=true random_id=3cb445cd-bf8b-f285-0a62-0a55a6cac7a7"
        }

      + identity {
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = "SystemAssigned"
        }
    }
```

### body content

The body does not contain any reference to `windowsProfile`:

```hcl
  properties_base = {
    addonProfiles          = local.addon_profiles
    agentPoolProfiles      = local.agent_pool_profiles
    apiServerAccessProfile = local.api_server_access_profile
    azureMonitorProfile    = local.monitor_profile
    diskEncryptionSetID    = var.disk_encryption_set_id
    kubernetesVersion      = var.kubernetes_version
    networkProfile         = local.network_profile_map
```

### Sensitive body content

Here we build a dynamic object based on the configuration. In this case `var.windows_profile` is `null`.

```hcl
locals {
  sensitive_body = {
    properties = {
      windowsProfile = var.windows_profile != null ? {
        adminPassword = var.windows_profile.admin_password
      } : null
    }
  }
}
```

Link to example that generates the error:

<https://github.com/Azure/terraform-azurerm-avm-res-containerservice-managedcluster/blob/9f50bb4b9b3a119ece94baa112efcb75865a7acd/examples/automatic>

/cc @pauldotyu

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## v2.8.0 (Unreleased)
 
+FEATURES:
+- **New Provider Function**: snake2camel
+
 ENHANCEMENTS:
 - `azapi_data_plane_resource` resource: Adds a customization layer which allows custom CRUD operations for resources that don't follow standard patterns. 
 - `azapi_data_plane_resource` resource: Support `Microsoft.KeyVault/vaults/keys` type.
@@ -8,7 +11,12 @@ ENHANCEMENTS:
 - `azapi_data_plane_resource` resource: Support `Microsoft.Search/searchServices/indexers` type.
 - `azapi_data_plane_resource` resource: Support `Microsoft.Search/searchServices/indexes` type.
 - `azapi_data_plane_resource` resource: Support `Microsoft.Search/searchServices/skillsets` type.
-- `azapi_data_plane_resource` resource: Support `Microsoft.Search/searchServices/synonymmaps` type. 
+- `azapi_data_plane_resource` resource: Support `Microsoft.Search/searchServices/synonymmaps` type.
+- Bump Go version to 1.24.6 to address CVEs (GH-992).
+- Add more verified `azapi` examples.
+
+BUG FIXES:
+- Fix panic when using `sensitive_body_version` with empty `sensitive_body` (GH-999). 
 
 ## v2.7.0
 

--- a/tools/generator-example-doc/main.go
+++ b/tools/generator-example-doc/main.go
@@ -221,6 +221,7 @@ func generateDocumentation(inputDir string) (string, error) {
 	// Read remarks.md if it exists (optional file)
 	remarksPath := path.Join(inputDir, "remarks.md")
 	remarks := ""
+	// #nosec G304 - remarksPath is constructed from controlled directory path
 	if remarksContent, err := os.ReadFile(remarksPath); err == nil {
 		remarks = strings.TrimSpace(string(remarksContent))
 	}

--- a/tools/sync-reference-doc/main.go
+++ b/tools/sync-reference-doc/main.go
@@ -148,6 +148,7 @@ func main() {
 		// Read optional remarks.md file from the resource folder
 		remarksPath := filepath.Join(resDir, "remarks.md")
 		remarksContent := ""
+		// #nosec G304 - remarksPath is constructed from controlled directory path
 		if remarksData, err := os.ReadFile(remarksPath); err == nil {
 			remarksContent = stripMarkdown(strings.TrimSpace(string(remarksData)))
 		}

